### PR TITLE
10.0/fuel-ui-fake: fake ui ci for fuel 10

### PR DESCRIPTION
**Pull request by rkhozinov**

### CODE REVIEW

**Review by naumvd95 — APPROVED** [review]

File: servers/partner-ci/10.0/ui-fake-nigthly.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+- ui-condition
+- multijob:
+    name: 'UI tests for {fuel_release}'
+    condition: SUCCESSFUL
+    projects:
+     - name: '{fuel_release}.{job_name}.test_nics'
+       property-file: build_number
+       current-parameters: true
+       predefined-parameters: |
+         TEST_PREFIX=test_nics*
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake-nigthly.yaml]: TEST_PATH (see issue)

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 2]
```diff
@@ -0,0 +1,234 @@
+---
+- parameter:
+    name: specific-parameters
+    parameters:
+      - text:
+          name: SPECIFIC_PROPS
+          default: |
+            FUEL_QA_TEST_GROUP=prepare_slaves_9;
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml]: fuel-qa is not needed

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+            UBUNTU_DIST=xenial;
+
+- project:
+    name: 'predefined_parameters'
+    fuel_release: '10.0'
+    fuel_release_checker: '10'
+    job_name: 'ui'
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml]: ui-fake

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    custom_name: '{fuel_release}.{job_name}.custom'
+    reporter-name: '{fuel_release}.{job_name}.reporter'
+    nightly_timer: 'H 21 * * *'
+    nightly_disabled: false
+    force_erase: false
+    force_reuse: true
+    recreate_venv: false
+    kvm_use: false
+    update_master: true
+    enable_proposed: true
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml]: fuel-related parameters are not needed (except fuel-ui)

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    debug: true
+    publish_results: true
+    fuel_ui_refs: ''
+    fuel_qa_refs: ''
+    fuel_qa_branch: 'origin/stable/newton'
+    fuel_ui_branch: 'origin/stable/newton'
+    docker_branch: 'master'
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml]: fake-ui

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    email_to: 'rkhozinov@mirantis.com'
+    scm_submodule_disabled: true
+    scm_wipe_workspace: false
+    skip_vote_successfull: false
+    skip_vote_failed: false
+
+    testrail_project: "Mirantis OpenStack"
+    testrail_suite: "[10.0][PCE] Mechanism for providing additional properties for NICs and Nodes via plugin"
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml]: [10.0][Fuel] UI testing

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+    testrail_run_desc: "Test run description"
+    testrail_item_case: "${{TEST_GROUP}}"
+    trep_results_source: "xunit"
+    trep_debug_level: "WARNING"
+
+    testgroup_parameter_type: string
+    testgroup_value_type: default
+    testgroup_value: '{testgroup}'
+
+    testgroup:
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml:52]: - test_cluster_workflows
- test_deployment_history

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+      - git:
+          url: 'http://git.openstack.org/openstack/fuel-ui'
+          basedir: fuel-ui
+          branches:
+            - $FUEL_UI_BRANCH
+          wipe-workspace: true
+          choosing-strategy: default
+          skip-tag: true
+      - git:
+          url: 'http://github.com/rkhozinov/fuel-ui-tests-docker'
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml]: ekhomyakova

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+      - specific-parameters
+
+- defaults:
+    name: global
+    disabled: false
+    node: ui
+    <<: *test-parameters
+    scm:
+      - git:
+          url: 'http://git.openstack.org/openstack/fuel-qa'
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml]: fuel-web instead

File: servers/partner-ci/10.0/ui-fake-nigthly.yaml (hunk 1 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake-nigthly.yaml]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 2 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 3 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 4 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 5 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 6 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 7 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml:52]: doen

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 9 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 8 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml]: done

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+    name: 'predefined_parameters'
+    fuel_release: '10.0'
+    fuel_release_checker: '10'
+    job_name: 'ui-fake'
+    nightly-name: '{fuel_release}.{job_name}.nightly'
+    custom_name: '{fuel_release}.{job_name}.custom'
+    reporter-name: '{fuel_release}.{job_name}.reporter'
+    nightly_timer: 'H 21 * * *'
+    nightly_disabled: false
+    force_erase: false
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml:20]: Seems force_erase and force_reuse will not do anything

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+    scm_submodule_disabled: true
+    scm_wipe_workspace: false
+    skip_vote_successfull: false
+    skip_vote_failed: false
+
+    testrail_project: "Mirantis OpenStack"
+    testrail_suite: "[10.0][Fuel] UI testing"
+    testrail_suite_desc: "Suite description"
+    testrail_milestone: "10.0"
+    testrail_plan: "${{TREP_TESTRAIL_MILESTONE}} iso #${{SNAPSHOTS_ID}}"
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml:40]: Change `iso #${{SNAPSHOTS_ID}}` to "ui" with timestamp, e.g. `ui 03/10/2017 18:30`

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+          publish_results: false
+          testgroup_parameter_type: choice
+          testgroup_value_type: choices
+          testgroup_value: '{obj:testgroup}'
+
+      - '{fuel_release}.{job_name}.{testgroup}':
+          scm_submodule_disabled: false
+
+      - '{reporter-name}':
+          plugin_version: ''
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml:75]: Seems it is not needed

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+          plugin_name: {job_name}
+
+- runner-parameters: &runner-parameters
+    name: 'runner-parameters'
+    parameters:
+      - common-fuel-parameters:
+          debug:                 '{debug}'
+          snapshots_id:           ''
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml:83]: Maybe snapshots_id, force_erase, force_reuse and fuel_release_checker are not needed

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+      - label:
+          name: node
+          default: 'ui'
+          description: "The node on which to run the job"
+      - '{testgroup_parameter_type}':
+          name: TEST_GROUP
+          '{testgroup_value_type}': '{obj:testgroup_value}'
+          description: "test-group name"
+      - string:
+          name: TEST_PREFIX
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml:116]: TEST_PATH?

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+          default: '$TEST_GROUP*'
+          description: "test-group name"
+      - common-fuel-parameters:
+          debug:                 '{debug}'
+          snapshots_id:           ''
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml:121]: not needed?

File: servers/partner-ci/10.0/ui-fake.yaml [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+          basedir: docker
+          branches:
+             - $DOCKER_BRANCH
+          wipe-workspace: true
+          skip-tag: true
+      - git:
+          url: 'http://github.com/openstack/fuel-web'
+          basedir: docker/fuel-web
+          branches:
+            - $FUEL_QA_BRANCH
```
Comment (ekhomyakova) [inline servers/partner-ci/10.0/ui-fake.yaml:159]: FUEL_WEB_BRANCH

File: servers/partner-ci/builders/ui.fake.test.sh [hunk 17]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash -e
+# activate bash xtrace for script
+[[ "${DEBUG}" == "true" ]] && set -x || set +x
+
+. "${VENV_PATH}/bin/activate"
+cd ${WORKSPACE}/docker/
+pip install --upgrade docker-compose
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/ui.fake.test.sh:7]: Is this under RECREATE_VENV condition? I suppose, it should be

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 10 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml:20]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 12 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml:75]: its stub for correct job-working

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 11 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml:40]: need to reconfigure reporting before

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 13 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml:83]: some of them removed, the other- marked as stubs

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 15 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml:121]: as above

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 16 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml:159]: done

File: servers/partner-ci/builders/ui.fake.test.sh (hunk 17 above)
Comment (naumvd95) [reply servers/partner-ci/builders/ui.fake.test.sh:7]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 16 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml:159]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 14 above)
Comment (naumvd95) [reply servers/partner-ci/10.0/ui-fake.yaml:116]: done

File: servers/partner-ci/10.0/ui-fake.yaml (hunk 11 above)
Comment (rkhozinov) [reply servers/partner-ci/10.0/ui-fake.yaml:40]: you could get that code from contrail jobs

**Pull request closed without merging**